### PR TITLE
Add generic return for useItem

### PR DESCRIPTION
**Pull request by yoojinko**
The useItem function has been updated with a generic type parameter T, which represents the itemType to retrieve and use. The return type of the function is determined based on the itemType. The function accepts additional arguments (...args) specific to the itemType.

Example usage:
```
useItem(...) // returns a Promise<void> | Promise<string> | Promise<NftMetadata>.
useItem<itemType.TICKET>(...) // returns a Promise<void>.
useItem<itemType.NFT_TRAIT>(...) // returns a Promise<NftMetadata>.
useItem<itemType.NFT>(...) // returns a Promise<string>.
```
cc: @ehgmsdk20

### CODE REVIEW

**Review by ehgmsdk20 — APPROVED** [review]
LGTM, thanks!

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

Please wait for @woomurf 's review.

File: src/store.ts [hunk 1]
```diff
@@ -520,21 +522,22 @@ export default class Store extends AinftBase {
...
    */
-  useItem({
+  useItem<T extends itemType>({
     appId,
     userId,
     itemName,
     quantity,
     params,
-  }: ItemUseParams): Promise<NftMetadata | null> {
+  }: ItemUseParams): UseItemReturnType[T] {
```
Comment (woomurf) [inline src/store.ts:540]: Could you add null type in this or NFT trait return type?

**Review by woomurf — APPROVED** [review]
LGTM with minor comments

File: src/store.ts (hunk 1 above)
Comment (yoojinko) [reply src/store.ts:540]: From my understanding, in the previous code, the return type null is returned when the itemType is not NFT_TRAIT (when it is TICKET). So I specified 'Promise<void>' for the TICKET type. 
Could you please confirm if my understanding is correct?

Comment (woomurf) [reply src/store.ts:540]: Oh. I understand it. Thanks!

**Comment by yoojinko** [comment]
Thanks for the reviews!!

**Comment by akaster99** [comment]
LGTM
Thanks!

**Pull request merged**